### ISSUE TITLE
Linked object fails to animate with bone parent

### ISSUE THREAD

**Issue by mysticfall** [UPBGE/upbge#1489]
When linking an object (`Ball` in my test case) and attaching it to an armature (`Cube`), it fails to move with the bone parent even though it works correctly in Blender itself.

In the attached project, `Ball` from `ball.blend` was linked from `cube-with-ball.blend` which moves with the `Cube` if you run the animation(`Action`) within Blender itself. However, if you launch it using the embedded UPBGE player, only the cube moves while the ball stays in its original position:

[override.tar.gz](https://github.com/UPBGE/upbge/files/6527926/override.tar.gz)

**Comment by mysticfall** [comment]
I've spent some time to test the example, and here's what I understand now:

* The symptom only happens with a linked object _without_ library override. However, enforcing library override for everything wouldn't be an ideal solution as it means the user must reposition and reparent all the relevant parts everytime such an object is added, which partly beats the purpose of using an asset library.
* Each copy of a linked object is duplicated again in runtime. It means, if I link an object and create two instances (i.e. collection instancing, previously known as "dupli group") of it, there'd be 4 objects in total while the game is running.
* If there's a component attached to the source object, only those attached to the final copy gets invoked in runtime. And in that case, the object's `groupObject` will refer to the corresponding source object (the instanced version).
* Playing an action, however, doesn't transform the final copy of the object, but its `groupObject` instead.
* As such, in order to make the example work as intended, an workaround as shown below (suggested by youle) is needed:
```python
# The source object (instanced from the linked source):
source = scene.objects["Ball"]

#groupMembers[0] refers to its runtime copy:
ball.groupMembers[0].worldTransform = worldTransform
```
* Playing the action from Blender works fine since there's no copies of instances involved. (Instances themselves animate just fine in UPBGE although invisible, hence the above workaround can work.)

I wonder if there's something can be done on the engine's side. If a copy should follow its source transform always, maybe we can do it automatically somehow?

I'm not sure why the behaviour of making a copy of each instance is needed. But if it is to be the case, I think it'd be better if we wouldn't have to apply such a workaround to every instance by hand.

**Comment by youle31** [comment]
It was automatically closed when i pushed my commit. As it's a complex thing, i can't guarantee the "fix" will work in all cases or won't cause any other issue, but I think it is not too dangerous and can be changed if you notice anomalies.

**Comment by mysticfall** [comment]
I just tested the new version and it indeed fixed the parenting problem for me. The linked child (a hair mesh, in my case) now animates with the parent object (the player character) now, but with a lag similar to what is reported in #1470.

However, I failed to reproduce the symptom with a simple test case, so I assume it could be just a side effect of the massive lag caused by trying to use particle hair in game.

For now, I'm pretty happy with the result as I think using library override to reuse assets has become much more practical than before. Thanks much for the work!
